src/Web/Controllers/AccountController.cs

```diff
@@ -103,120 +101,166 @@ public async Task<ActionResult> SignOut()
+                PhoneNumber = model.Phone,
+                Street1 = model.Address1,
+                Street2 = model.Address2,
+                City = model.City,
+                State = model.State,
+                ZipCode = model.ZipCode,
+                Country = model.CountryRegion
+            };
+
+            var signupUser = new SignupUser
+            {
+                CreationDate = DateTimeOffset.Now,
+                UpdateDate = DateTimeOffset.Now,
+                FirstName = model.FirstName,
+                LastName = model.LastName,
+                Country = model.CountryRegion,
+                ZipCode = model.ZipCode,
+                TimeZone = new TimeSpan(0, 0, -model.ClientTimeZone, 0).ToString(),
+                Language = "EN-US",
+                PhoneNumber = model.Phone,
+                Email = model.Email,
+                ProfileType = UserProfileType.Consumer.ToString(),
+                Address = signupAddress
+            };
+
+            var deviceContext = new DeviceContext
+            {
+                DeviceContextId = _contextAccessor.GetSessionId(),
+                IPAddress = _contextAccessor.HttpContext.Connection.RemoteIpAddress.ToString(),
+                DeviceContextDC = model.DeviceFingerPrinting.FingerPrintingDC,
+                Provider = DeviceContextProvider.DFPFingerPrinting.ToString(),
+            };
+
+            var marketingContext = new MarketingContext
+            {
+                Type = MarketingType.Direct.ToString(),
+                IncentiveType = MarketingIncentiveType.None.ToString(),
+                IncentiveOffer = "Integrate with Fraud Protection"
+            };
+
+            var storefrontContext = new StoreFrontContext
+            {
+                StoreName = "Fraud Protection Sample Site",
+                Type = StorefrontType.Web.ToString(),
+                Market = "US"
+            };
 
+            var signupEvent = new SignUp
+            {
+                SignUpId = Guid.NewGuid().ToString(),
+                AssessmentType = AssessmentType.Protect.ToString(),
+                User = signupUser,
+                MerchantLocalDate = DateTimeOffset.Now,
+                CustomerLocalDate = model.ClientDate,
+                MarketingContext = marketingContext,
+                StoreFrontContext = storefrontContext,
+                DeviceContext = deviceContext,
+            };
+
+            var correlationId = _fraudProtectionService.NewCorrelationId;
+
+            var signupAssessment = await _fraudProtectionService.PostSignup(signupEvent, correlationId);
+
+            //Track Fraud Protection request/response for display only
+            var fraudProtectionIO = new FraudProtectionIOModel(signupEvent, signupAssessment, "Signup");
+
+            //2 out of 3 signups will succeed on average. Adjust if you want more or less signups blocked for tesing purposes.
+            var random = new Random();
```

Comment:
nit: only the next line uses random here, can't we do new Random().NextDouble()?

Reply:
Further more, can't we just use:

new Random().Next(0, 3) !== 0

?

Reply:
Yeah, that's a nice cleanup.  FYI, I went 0-2 so that it's still 1/3 rather than 1/4. 

```new Random().Next(0, 2) != 0;```